### ISSUE TITLE
Ability to decorate `*Fixture` and `*ExampleFactory`

### ISSUE THREAD

**Issue by igormukhingmailcom** [Sylius/Sylius#10252]
**TLDR Context**
When implementing plugins for Sylius, I usually add fixtures - that way app developer can "in one click" try plugin (without eating time to add some data manually).

Sometime plugins extending core resources (Product, Order, etc) and I believe we should have **safe** way to extend or even decorate `*ExampleFactory` for core resources. 

**Safe** means that solution should work even if 2 or more plugins will be installed that make changes to **same** `*ExampleFactory`. This can be done with decorators I believe.

But methods at `*ExampleFactory` and `*Factory` protected which make that impossible to create **safe** solutions.

**Describe the proposed solution**

1. Replace:

```php
protected function configureResourceNode(ArrayNodeDefinition $resourceNode): void;
```

with

```php
public function configureResourceNode(ArrayNodeDefinition $resourceNode): void;
```

at `AbstractResourceFixture` and related core resources fixtures.

2. Add `ConfigurableExampleFactoryInterface` (or `DecorableExampleFactoryInterface`, not sure about name) - optional

```php
interface ConfigurableExampleFactoryInterface extends ExampleFactoryInterface
{
    public function configureOptions(OptionsResolver $resolver): void;
}
```

3. Replace `AbstractExampleFactory` with:

```php
abstract class AbstractExampleFactory implements ConfigurableExampleFactoryInterface
{
    abstract public function configureOptions(OptionsResolver $resolver): void;
}
```

**Additional context**
As far as currently we have some BC breaking changes in this area ("Make FixturesBundle standalone") - I believe this is good time to do proposed change too.

I will make PR - just let me know please to what branch I should base it to.

**Comment by loevgaard** [comment]
What's your take on this, @pamil, @Zales0123? :)

**Comment by igormukhingmailcom** [comment]
Hi, @pamil .
Noticed you guys released 1.5 with extracted fixture bundle. 
What about this small but very important change (now probably at `1.6`)? 
With it - we can override fixtures with decorators - that way different plugin's fixtures overrides will not conflict with each other.
Thanks :)

**Comment by vvasiloi** [comment]
Until any changes to facilitate this process are made, I will leave here a few tips I use to decorate both **Fixture** and **ExampleFactory** classes.

**Docorating ExampleFactory** classes: 
https://gist.github.com/vvasiloi/a3a8a7b151775bb58cd122a47b2cc4dd

 **Decorating Fixture** classes:
since these implement the `Symfony\Component\Config\Definition\ConfigurationInterface`, you have the public method `getConfigTreeBuilder` which can be used to make any modification you want to the configuration of the decorated class.
The `Sylius\Bundle\FixturesBundle\Fixture\FixtureInterface::load` public method is a bit harder to work with, but most of the time you don't need to change what it does because it's mainly about bulk processing and persistence, while the actual object creation is delegated to the factories.
In the worst case you will have to override it completely (not call the decorated method).

**Comment by igormukhingmailcom** [comment]
Do not stale

**Comment by lchrusciel** [comment]
@igormukhingmailcom there are attempts to make it better like #10616. But the core of the issue is related to https://github.com/Sylius/SyliusFixtureBundle, can we move this conversation there?
